In [1]:
import numpy as np
import matplotlib.pyplot as plt
from model import *

In [2]:
# model = MoneyModel(5, 10, 10)
# for i in range(20):
#     model.step()

In [3]:
# agent_counts = np.zeros((model.grid.width, model.grid.height))
# for cell in model.grid.coord_iter():
#     cell_content, x, y = cell
#     agent_count = len(cell_content)
#     agent_counts[x][y] = agent_count
# plt.imshow(agent_counts, interpolation="nearest")
# plt.colorbar()

In [4]:
empty_model = MyModel(4, 2, 120, 10, 10)
for i in range(100):
    empty_model.step()

# if len(model.agents) != 0:
#     for player in model.agents:
#         print("Player", player.unique_id, "age:", player.age)


Step: 0
Club 0. My type is: 2. My team has player number: 10, 11, 19, 20, 21, 28, 31, 36, 49, 55, 56, 63, 75, 78, 85, 90, 98, 100, 101, 103, 104, 105, 112, 115, 118, 119, 123, 124, 125, 126. The average level is: 5.9. The revenue is 138. The number of fan is 69. Spending is 103.16. The budget is 34.84.
Club 3. My type is: 3. My team has player number: 22, 30, 41, 46, 48, 51, 54, 57, 61, 77, 80, 82, 83, 86, 87, 88, 89, 91, 96, 102, 107, 109, 110, 111, 114, 116, 117, 120, 121, 122. The average level is: 5.23. The revenue is 50. The number of fan is 25. Spending is 82.67. The budget is -32.67.
Club 2. My type is: 3. My team has player number: 8, 9, 12, 16, 17, 18, 24, 25, 29, 32, 35, 38, 39, 40, 42, 43, 44, 45, 50, 53, 60, 64, 65, 67, 68, 69, 72, 76, 81, 94. The average level is: 5.17. The revenue is 36. The number of fan is 18. Spending is 104.16. The budget is -68.16.
Club 1. My type is: 1. My team has player number: 7, 13, 14, 15, 23, 26, 27, 33, 34, 37, 47, 52, 58, 59, 62, 66, 70, 71

My agent is agent 5. My club is club 2
Hi, I am player 59. My age is 22. My contract is Signed. My rep is 6. My skill is 3. My value is 2.45 millions. My salary is 3.27.
My agent is agent 5. My club is club 1
Hi, I am player 43. My age is 38. My contract is Signed. My rep is 6. My skill is 7. My value is 14.37 millions. My salary is 4.42.
My agent is agent 5. My club is club 2
Hi, I am player 79. My age is 24. My contract is Signed. My rep is 8. My skill is 4. My value is 14.67 millions. My salary is 5.33.
My agent is agent 5. My club is club 1
Hi, I am player 10. My age is 36. My contract is Signed. My rep is 2. My skill is 9. My value is 8.0 millions. My salary is 1.0.
My agent is agent 4. My club is club 0
Hi, I am player 102. My age is 35. My contract is Signed. My rep is 1. My skill is 7. My value is 1.4 millions. My salary is 0.4.
My agent is agent 4. My club is club 3
Hi, I am player 120. My age is 20. My contract is Signed. My rep is 2. My skill is 4. My value is 5.2 millions. 

My agent is agent 4. My club is club 1
Hi, I am player 12. My age is 34. My contract is Signed. My rep is 8. My skill is 12. My value is 56.47 millions. My salary is 5.65.
My agent is agent 4. My club is club 2
Hi, I am player 123. My age is 24. My contract is Signed. My rep is 6. My skill is 12. My value is 54.0 millions. My salary is 12.0.
My agent is agent 5. My club is club 0
Hi, I am player 72. My age is 29. My contract is Signed. My rep is 5. My skill is 9. My value is 17.07 millions. My salary is 6.21.
My agent is agent 5. My club is club 2
Hi, I am player 86. My age is 24. My contract is Signed. My rep is 6. My skill is 5. My value is 16.25 millions. My salary is 5.0.
My agent is agent 5. My club is club 3
Hi, I am player 130. My age is 33. My contract is Signed. My rep is 9. My skill is 7. My value is 30.55 millions. My salary is 3.82.
My agent is agent 4. My club is club 3
Hi, I am player 79. My age is 25. My contract is Signed. My rep is 8. My skill is 5. My value is 17.6 mi

Hi, I am player 106. My age is 22. My contract is Signed. My rep is 6. My skill is 6. My value is 16.36 millions. My salary is 6.55.
My agent is agent 5. My club is club 1
Hi, I am player 133. My age is 22. My contract is Free agent. My rep is 4. My skill is 1. My value is 0.73 millions. My salary is 0.36.
Hi, I am player 47. My age is 23. My contract is Signed. My rep is 7. My skill is 6. My value is 10.96 millions. My salary is 3.65.
My agent is agent 4. My club is club 1
Hi, I am player 28. My age is 34. My contract is Signed. My rep is 7. My skill is 8. My value is 16.47 millions. My salary is 6.59.
My agent is agent 5. My club is club 0
Hi, I am player 77. My age is 33. My contract is Signed. My rep is 3. My skill is 7. My value is 3.39 millions. My salary is 1.27.
My agent is agent 4. My club is club 3
Hi, I am player 131. My age is 35. My contract is Signed. My rep is 1. My skill is 3. My value is 0.51 millions. My salary is 0.17.
My agent is agent 4. My club is club 0
Hi, I am 

My agent is agent 4. My club is club 1
Hi, I am player 54. My age is 27. My contract is Free agent. My rep is 5. My skill is 6. My value is 8.89 millions. My salary is 2.22.
Hi, I am player 94. My age is 24. My contract is Signed. My rep is 11. My skill is 11. My value is 65.54 millions. My salary is 10.08.
My agent is agent 4. My club is club 2
Hi, I am player 134. My age is 32. My contract is Free agent. My rep is 5. My skill is 4. My value is 2.5 millions. My salary is 2.5.
Hi, I am player 42. My age is 28. My contract is Signed. My rep is 7. My skill is 11. My value is 30.25 millions. My salary is 5.5.
My agent is agent 4. My club is club 2
Hi, I am player 100. My age is 34. My contract is Signed. My rep is 8. My skill is 6. My value is 9.88 millions. My salary is 2.82.
My agent is agent 4. My club is club 0
Hi, I am player 46. My age is 30. My contract is Signed. My rep is 8. My skill is 7. My value is 13.07 millions. My salary is 7.47.
My agent is agent 5. My club is club 1
Hi, I

Hi, I am player 42. My age is 29. My contract is Signed. My rep is 7. My skill is 11. My value is 29.21 millions. My salary is 5.31.
My agent is agent 4. My club is club 2
Hi, I am player 65. My age is 39. My contract is Free agent. My rep is 4. My skill is 1. My value is 0.1 millions. My salary is 0.41.
Hi, I am player 47. My age is 26. My contract is Signed. My rep is 7. My skill is 6. My value is 9.69 millions. My salary is 3.23.
My agent is agent 4. My club is club 1
Hi, I am player 142. My age is 20. My contract is Signed. My rep is 1. My skill is 5. My value is 2.5 millions. My salary is 1.0.
My agent is agent 5. My club is club 0
Hi, I am player 109. My age is 38. My contract is Signed. My rep is 6. My skill is 3. My value is 1.42 millions. My salary is 1.89.
My agent is agent 5. My club is club 0
Hi, I am player 25. My age is 33. My contract is Signed. My rep is 7. My skill is 7. My value is 16.33 millions. My salary is 5.94.
My agent is agent 5. My club is club 0
Hi, I am play

Hi, I am player 100. My age is 36. My contract is Signed. My rep is 8. My skill is 7. My value is 10.89 millions. My salary is 3.11.
My agent is agent 4. My club is club 0
Hi, I am player 28. My age is 38. My contract is Signed. My rep is 7. My skill is 10. My value is 18.42 millions. My salary is 7.37.
My agent is agent 5. My club is club 0
Hi, I am player 163. My age is 26. My contract is Signed. My rep is 8. My skill is 2. My value is 1.85 millions. My salary is 2.46.
My agent is agent 5. My club is club 3
Hi, I am player 64. My age is 32. My contract is Free agent. My rep is 1. My skill is 2. My value is 0.12 millions. My salary is 0.25.
Hi, I am player 82. My age is 38. My contract is Signed. My rep is 5. My skill is 5. My value is 3.29 millions. My salary is 1.32.
My agent is agent 4. My club is club 1
Hi, I am player 106. My age is 26. My contract is Signed. My rep is 6. My skill is 10. My value is 23.08 millions. My salary is 9.23.
My agent is agent 5. My club is club 1
Hi, I a

Hi, I am player 58. My age is 28. My contract is Signed. My rep is 8. My skill is 6. My value is 10.29 millions. My salary is 3.43.
My agent is agent 4. My club is club 1
Hi, I am player 105. My age is 39. My contract is Signed. My rep is 11. My skill is 13. My value is 51.33 millions. My salary is 14.67.
My agent is agent 5. My club is club 0
Hi, I am player 101. My age is 37. My contract is Signed. My rep is 6. My skill is 7. My value is 7.95 millions. My salary is 4.54.
My agent is agent 5. My club is club 0
Hi, I am player 89. My age is 38. My contract is Signed. My rep is 7. My skill is 10. My value is 18.42 millions. My salary is 3.68.
My agent is agent 4. My club is club 3
Agent 5 My clients are: 20, 25, 39, 46, 48, 53, 59, 64, 67, 69, 72, 74, 79, 83, 86, 91, 92, 101, 105, 106, 108, 113, 117, 120, 123, 127, 132, 134, 138, 140, 141, 142, 145, 148, 150, 152, 153, 154, 155, 156, 158, 161, 163, 167, 168, 169, 170, 173, 174, 178, 181, 182, 11. My skill is 4. 
Agent 5 money is 212.13.

Hi, I am player 22. My age is 35. My contract is Signed. My rep is 12. My skill is 8. My value is 21.94 millions. My salary is 5.49.
My agent is agent 4. My club is club 3
Hi, I am player 183. My age is 25. My contract is Signed. My rep is 4. My skill is 9. My value is 12.96 millions. My salary is 2.88.
My agent is agent 4. My club is club 1
Hi, I am player 25. My age is 38. My contract is Signed. My rep is 7. My skill is 11. My value is 22.29 millions. My salary is 8.11.
My agent is agent 5. My club is club 0
Hi, I am player 166. My age is 29. My contract is Signed. My rep is 7. My skill is 12. My value is 40.55 millions. My salary is 5.79.
My agent is agent 4. My club is club 3
Hi, I am player 164. My age is 37. My contract is Free agent. My rep is 2. My skill is 1. My value is 0.05 millions. My salary is 0.11.
Hi, I am player 108. My age is 39. My contract is Signed. My rep is 4. My skill is 16. My value is 26.26 millions. My salary is 6.56.
My agent is agent 5. My club is club 1
Hi

My agent is agent 4. My club is club 3
Hi, I am player 190. My age is 38. My contract is Signed. My rep is 3. My skill is 8. My value is 8.21 millions. My salary is 2.53.
My agent is agent 5. My club is club 0
Hi, I am player 154. My age is 32. My contract is Free agent. My rep is 1. My skill is 8. My value is 2.0 millions. My salary is 1.0.
Hi, I am player 69. My age is 38. My contract is Signed. My rep is 10. My skill is 6. My value is 9.47 millions. My salary is 6.32.
My agent is agent 5. My club is club 0
Hi, I am player 94. My age is 31. My contract is Signed. My rep is 13. My skill is 13. My value is 70.87 millions. My salary is 10.9.
My agent is agent 4. My club is club 2
Hi, I am player 117. My age is 39. My contract is Signed. My rep is 12. My skill is 12. My value is 44.31 millions. My salary is 14.77.
My agent is agent 5. My club is club 3
Hi, I am player 120. My age is 31. My contract is Signed. My rep is 9. My skill is 13. My value is 49.06 millions. My salary is 15.1.
My 

My agent is agent 5. My club is club 3
Hi, I am player 197. My age is 19. My contract is Signed. My rep is 11. My skill is 11. My value is 114.63 millions. My salary is 12.74.
My agent is agent 4. My club is club 2
Hi, I am player 173. My age is 33. My contract is Signed. My rep is 6. My skill is 10. My value is 23.64 millions. My salary is 7.27.
My agent is agent 5. My club is club 0
Hi, I am player 44. My age is 32. My contract is Signed. My rep is 14. My skill is 9. My value is 35.44 millions. My salary is 7.88.
My agent is agent 4. My club is club 2
Hi, I am player 128. My age is 31. My contract is Signed. My rep is 5. My skill is 11. My value is 19.52 millions. My salary is 3.55.
My agent is agent 4. My club is club 2
Hi, I am player 199. My age is 36. My contract is Signed. My rep is 1. My skill is 9. My value is 2.25 millions. My salary is 1.0.
My agent is agent 5. My club is club 3
Hi, I am player 133. My age is 32. My contract is Free agent. My rep is 4. My skill is 4. My valu

Hi, I am player 171. My age is 29. My contract is Signed. My rep is 1. My skill is 16. My value is 11.03 millions. My salary is 1.1.
My agent is agent 4. My club is club 0
Hi, I am player 213. My age is 22. My contract is Signed. My rep is 8. My skill is 4. My value is 14.55 millions. My salary is 5.82.
My agent is agent 5. My club is club 3
No players retire!
Close
Club 0 spends 126.89.
Club 1 spends 123.67.
Club 2 spends 174.26.
Club 3 spends 232.86.

Step: 59
<class 'agents.F_Agents'>
17
Before signing - player: <agents.Players object at 0x7f8213b20610>
Agent 4 signs player 17
<class 'agents.F_Agents'>
13
Before signing - player: <agents.Players object at 0x7f8213b206d0>
Agent 5 signs player 13
Hi, I am player 92. My age is 37. My contract is Signed. My rep is 1. My skill is 14. My value is 5.3 millions. My salary is 1.51.
My agent is agent 5. My club is club 1
Hi, I am player 81. My age is 37. My contract is Signed. My rep is 15. My skill is 15. My value is 91.22 millions. My salar

Hi, I am player 183. My age is 29. My contract is Signed. My rep is 4. My skill is 9. My value is 11.17 millions. My salary is 2.48.
My agent is agent 4. My club is club 1
Hi, I am player 42. My age is 38. My contract is Signed. My rep is 12. My skill is 11. My value is 38.21 millions. My salary is 6.95.
My agent is agent 4. My club is club 2
Hi, I am player 58. My age is 34. My contract is Signed. My rep is 8. My skill is 6. My value is 8.47 millions. My salary is 2.82.
My agent is agent 4. My club is club 1
Hi, I am player 211. My age is 29. My contract is Signed. My rep is 1. My skill is 3. My value is 1.14 millions. My salary is 0.21.
My agent is agent 4. My club is club 3
Hi, I am player 208. My age is 33. My contract is Signed. My rep is 5. My skill is 4. My value is 3.64 millions. My salary is 2.42.
My agent is agent 5. My club is club 1
Hi, I am player 59. My age is 36. My contract is Signed. My rep is 6. My skill is 3. My value is 1.5 millions. My salary is 2.0.
My agent is ag

Hi, I am player 63. My age is 34. My contract is Signed. My rep is 10. My skill is 16. My value is 75.29 millions. My salary is 9.41.
My agent is agent 4. My club is club 0
Hi, I am player 35. My age is 36. My contract is Signed. My rep is 13. My skill is 14. My value is 70.78 millions. My salary is 10.11.
My agent is agent 4. My club is club 2
Hi, I am player 236. My age is 23. My contract is Signed. My rep is 6. My skill is 3. My value is 3.13 millions. My salary is 3.13.
My agent is agent 5. My club is club 1
Hi, I am player 194. My age is 38. My contract is Signed. My rep is 6. My skill is 13. My value is 26.68 millions. My salary is 8.21.
My agent is agent 5. My club is club 2
Hi, I am player 123. My age is 38. My contract is Signed. My rep is 6. My skill is 18. My value is 51.16 millions. My salary is 11.37.
My agent is agent 5. My club is club 0
Hi, I am player 168. My age is 30. My contract is Signed. My rep is 1. My skill is 9. My value is 2.7 millions. My salary is 1.2.
My ag

Hi, I am player 192. My age is 38. My contract is Signed. My rep is 7. My skill is 11. My value is 22.29 millions. My salary is 4.05.
My agent is agent 4. My club is club 0
Hi, I am player 244. My age is 35. My contract is Signed. My rep is 9. My skill is 2. My value is 3.6 millions. My salary is 1.03.
My agent is agent 4. My club is club 2
Hi, I am player 140. My age is 39. My contract is Signed. My rep is 13. My skill is 16. My value is 85.33 millions. My salary is 21.33.
My agent is agent 5. My club is club 2
Hi, I am player 166. My age is 35. My contract is Signed. My rep is 8. My skill is 14. My value is 44.8 millions. My salary is 6.4.
My agent is agent 4. My club is club 3
Hi, I am player 215. My age is 34. My contract is Signed. My rep is 8. My skill is 7. My value is 11.53 millions. My salary is 6.59.
My agent is agent 5. My club is club 0
Hi, I am player 167. My age is 35. My contract is Signed. My rep is 11. My skill is 13. My value is 53.11 millions. My salary is 16.34.
My 

Hi, I am player 248. My age is 37. My contract is Signed. My rep is 3. My skill is 8. My value is 9.08 millions. My salary is 1.3.
My agent is agent 4. My club is club 1
Hi, I am player 120. My age is 37. My contract is Signed. My rep is 9. My skill is 13. My value is 41.11 millions. My salary is 12.65.
My agent is agent 5. My club is club 3
Hi, I am player 143. My age is 34. My contract is Signed. My rep is 2. My skill is 9. My value is 4.76 millions. My salary is 1.06.
My agent is agent 4. My club is club 0
Hi, I am player 258. My age is 29. My contract is Signed. My rep is 9. My skill is 9. My value is 33.52 millions. My salary is 11.17.
My agent is agent 5. My club is club 2
Hi, I am player 252. My age is 31. My contract is Signed. My rep is 4. My skill is 7. My value is 9.03 millions. My salary is 3.61.
My agent is agent 5. My club is club 1
Hi, I am player 208. My age is 36. My contract is Signed. My rep is 5. My skill is 6. My value is 5.0 millions. My salary is 3.33.
My agent i

My agent is agent 4. My club is club 0
Hi, I am player 240. My age is 33. My contract is Signed. My rep is 9. My skill is 7. My value is 24.82 millions. My salary is 3.82.
My agent is agent 4. My club is club 2
Hi, I am player 160. My age is 35. My contract is Signed. My rep is 4. My skill is 9. My value is 9.26 millions. My salary is 2.06.
My agent is agent 4. My club is club 1
Hi, I am player 13. My age is 38. My contract is Signed. My rep is 2. My skill is 12. My value is 7.58 millions. My salary is 1.26.
My agent is agent 5. My club is club 1
Hi, I am player 94. My age is 39. My contract is Signed. My rep is 21. My skill is 13. My value is 91.0 millions. My salary is 14.0.
My agent is agent 4. My club is club 2
Hi, I am player 269. My age is 27. My contract is Signed. My rep is 3. My skill is 3. My value is 1.0 millions. My salary is 0.67.
My agent is agent 4. My club is club 1
Hi, I am player 196. My age is 35. My contract is Signed. My rep is 8. My skill is 5. My value is 5.71 mi

My agent is agent 5. My club is club 2
Hi, I am player 216. My age is 26. My contract is Signed. My rep is 2. My skill is 9. My value is 6.92 millions. My salary is 1.38.
My agent is agent 4. My club is club 1
Hi, I am player 274. My age is 24. My contract is Signed. My rep is 9. My skill is 2. My value is 1.5 millions. My salary is 1.5.
My agent is agent 4. My club is club 1
Hi, I am player 280. My age is 24. My contract is Signed. My rep is 1. My skill is 8. My value is 4.0 millions. My salary is 1.33.
My agent is agent 5. My club is club 1
Hi, I am player 193. My age is 35. My contract is Signed. My rep is 6. My skill is 8. My value is 10.97 millions. My salary is 2.74.
My agent is agent 4. My club is club 1
Hi, I am player 235. My age is 29. My contract is Signed. My rep is 1. My skill is 14. My value is 8.21 millions. My salary is 1.93.
My agent is agent 5. My club is club 0
Hi, I am player 188. My age is 29. My contract is Signed. My rep is 11. My skill is 11. My value is 45.9 mi

Hi, I am player 294. My age is 26. My contract is Signed. My rep is 9. My skill is 3. My value is 3.12 millions. My salary is 2.08.
My agent is agent 4. My club is club 0
Hi, I am player 291. My age is 36. My contract is Signed. My rep is 8. My skill is 3. My value is 3.33 millions. My salary is 1.33.
My agent is agent 4. My club is club 2
Hi, I am player 270. My age is 25. My contract is Signed. My rep is 8. My skill is 9. My value is 25.92 millions. My salary is 11.52.
My agent is agent 5. My club is club 0
Hi, I am player 200. My age is 39. My contract is Signed. My rep is 3. My skill is 17. My value is 22.23 millions. My salary is 5.23.
My agent is agent 5. My club is club 3
Hi, I am player 302. My age is 21. My contract is Signed. My rep is 7. My skill is 3. My value is 11.0 millions. My salary is 2.0.
My agent is agent 4. My club is club 0
Hi, I am player 260. My age is 27. My contract is Signed. My rep is 9. My skill is 4. My value is 5.33 millions. My salary is 5.33.
My agent i

Hi, I am player 328. My age is 18. My contract is Signed. My rep is 3. My skill is 5. My value is 5.0 millions. My salary is 3.33.
My agent is agent 5. My club is club 0
Hi, I am player 310. My age is 25. My contract is Signed. My rep is 8. My skill is 5. My value is 12.8 millions. My salary is 3.2.
My agent is agent 4. My club is club 1
Hi, I am player 270. My age is 26. My contract is Signed. My rep is 8. My skill is 9. My value is 24.92 millions. My salary is 11.08.
My agent is agent 5. My club is club 0
Hi, I am player 296. My age is 33. My contract is Signed. My rep is 7. My skill is 4. My value is 3.39 millions. My salary is 1.7.
My agent is agent 4. My club is club 0
Hi, I am player 329. My age is 25. My contract is Signed. My rep is 6. My skill is 9. My value is 19.44 millions. My salary is 4.32.
My agent is agent 4. My club is club 3
Hi, I am player 280. My age is 27. My contract is Signed. My rep is 1. My skill is 11. My value is 4.89 millions. My salary is 1.63.
My agent is 

0: 5, 7, 11, 14    The revenue is 138. The number of fan is 69. Spending is 17.31. The budget is 120.69.
1: 6, 8, 9, 10, 12, 13, 15.     The revenue is 86. The number of fan is 43. Spending is 6.34. The budget is 79.66.
5: value: 0.25 salary: 1.0

In [5]:
# # from data import DataCollector

# # Create an instance of the model
# empty_model = MyModel(4, 2, 12, 10, 10)

# # Create a data collector
# # data_collector = DataCollector(model=empty_model)

# # Run the simulation and collect data
# for i in range(10):
#     empty_model.step()
#     data_collector.collect()

# # Access the collected data
# collected_data = data_collector.get_model_vars_dataframe()


In [6]:
from agents import Club, Players, F_Agents
from model import MyModel

# Create a model
model = MyModel(C=2, F=1, P=1, width=10, height=10)
model.open_market()

# Create a club
club = Club(unique_id=0, model=model, type=1)
club2 = Club(unique_id=1, model=model, type=2)

# Create an agent
agent = F_Agents(unique_id=2, model=model, cut=5, network="Signed", n_skills=5)

# Create a player
player1 = Players(unique_id=3, model=model, age=20, contract="Signed", reputation=5, skill=7)
player1.link_agent(agent)
player1.join_club(club)
player1.step()

print(player1)
print(club.team)
# Set the budget and spending of the club
club.budget = -1000
club.set_spending()

# Test the sell_player method
club.sell_player()

# Print the updated state of club and player
# print("Player state after selling:")
# print("Player agent:", player1.F_agent)
# print("Player club:", player1.club)



Before transfer - market: [<agents.Players object at 0x7f8213b6d040>]
Hi, I am player 3. My age is 20. My contract is Signed. My rep is 5. My skill is 7. My value is 0 millions. My salary is 0.
My agent is agent 2. My club is club 0
Min 3
Club 0 bought player 3
Club 0 sold player  3
Before transfer - player: <agents.Players object at 0x7f8213b43ca0>


In [7]:
from agents import Club, Players, F_Agents
from model import MyModel

# Create a model
model = MyModel(C=2, F=1, P=1, width=10, height=10)
model.step()

# Create a club
club = Club(unique_id=0, model=model, type=1)
print(club)

club2 = Club(unique_id=1, model=model, type=2)

# Create an agent
agent = F_Agents(unique_id=2, model=model, cut=5, network="Signed", n_skills=5)

# Create a player
player1 = Players(unique_id=3, model=model, age=20, contract="Signed", reputation=5, skill=7)
player1.link_agent(agent)
player1.join_club(club)
player1.step()

print("Initial state:")
print("Player:", player1)
print("Player club:", player1.club)
print("Club 1:", club1.team)
print("Club 2:", club2.team)


# Set the budget and spending of the club
club.budget = -1000
club.set_spending()

# Test the sell_player method
club.sell_player()

# Print the updated state of club and player
print("\nPlayer state after selling:")
print("Player:", player1)
print("Player club:", player1.club)
print("Club 1:", club1.team)
print("Club 2:", club2.team)



Step: 0
No players in the team yet
Club 0. My type is: 3. My team has player number: . The average level is: None. The revenue is 32. The number of fan is 16. Spending is 0. The budget is 32.
Club 1. My type is: 1. My team has player number: 4. The average level is: 7.0. The revenue is 156. The number of fan is 78. Spending is 6.81. The budget is 149.19.
Agent 2 My clients are: 4. My skill is 4. 
Agent 2 money is 0.2.
Hi, I am player 4. My age is 37. My contract is Signed. My rep is 9. My skill is 7. My value is 25.54 millions. My salary is 6.81.
My agent is agent 2. My club is club 1
No players retire!
Hi, I am player 3. My age is 20. My contract is Signed. My rep is 5. My skill is 7. My value is 0 millions. My salary is 0.
My agent is agent 2. My club is club 0
Initial state:
Player: <agents.Players object at 0x7f8213b6dc10>
Player club: <agents.Club object at 0x7f8213b6d6a0>


NameError: name 'club1' is not defined

In [ ]:
from agents import Club, Players, F_Agents
from model import MyModel

# Create a model
model = MyModel(C=2, F=1, P=1, width=10, height=10)

# Create two clubs
club1 = Club(unique_id=0, model=model, type=1)
club2 = Club(unique_id=1, model=model, type=2)

# Create an agent
agent = F_Agents(unique_id=2, model=model, cut=5, network="Signed", n_skills=5)

# Create a player and add them to club1
player1 = Players(unique_id=3, model=model, age=20, contract="Signed", reputation=5, skill=7)
player1.link_agent(agent)
player1.join_club(club1)
player1.set_value()
player1.set_salary()

# Print initial state of player and clubs
print("Initial state:")
print("Player:", player1)
print("Player club:", player1.club)
print("Club 1:", club1.team)
print("Club 2:", club2.team)

model.open_market()

# model.transfer(player1, club2)

# Set the budget and spending of club1
club1.budget = -1000
club2.budget = 2000
# club1.set_spending()

# Sell the player from club1 to club2
print("Player club:", player1.club)
club1.sell_player()

# Print the updated state of player and clubs after selling
print("\nPlayer state after selling:")
print("Player:", player1)
print("Player agent:", player1.F_agent)
print("Player club:", player1.club)
print("Club 1:", club1.team)
print("Club 2:", club2.team)


In [ ]:
free = []
free_agents = [player for player in free]
print(type(free_agents))

In [8]:
# Import the required modules
from mesa import Model
from agents import *
from mesa.space import MultiGrid
from mesa.time import RandomActivationByType
import random

# Define the number of clubs, agents, players, width, and height
num_clubs = 5
num_agents = 3
num_players = 10
width = 10
height = 10

# Create an instance of the model
model = MyModel(num_clubs, num_agents, num_players, width, height)

# Run the model for a few steps (for example, 5 steps)
for _ in range(5):
    model.step()

# Call the agent_incentives() method
model.agent_incentives()



Step: 0
Club 3. My type is: 1. My team has player number: 9, 10. The average level is: 3.0. The revenue is 162. The number of fan is 81. Spending is 2.82. The budget is 159.18.
Club 0. My type is: 3. My team has player number: 12, 14. The average level is: 9.0. The revenue is 52. The number of fan is 26. Spending is 11.0. The budget is 41.0.
Club 4. My type is: 2. My team has player number: 17, 18. The average level is: 4.0. The revenue is 66. The number of fan is 33. Spending is 9.39. The budget is 56.61.
Club 2. My type is: 2. My team has player number: 11, 13. The average level is: 7.0. The revenue is 106. The number of fan is 53. Spending is 12.96. The budget is 93.04.
Club 1. My type is: 1. My team has player number: 15, 16. The average level is: 3.0. The revenue is 168. The number of fan is 84. Spending is 8.79. The budget is 159.21.
Hi, I am player 15. My age is 21. My contract is Signed. My rep is 5. My skill is 4. My value is 9.52 millions. My salary is 6.67.
My agent is agen

ValueError: min() arg is an empty sequence